File: Source/Bifrost.JavaScript/specifications/Or.js
Bifrost.namespace("Bifrost.specifications", {
    Or: Bifrost.specifications.Specification.extend(function (leftHandSide, rightHandSide) {
        /// <summary>Represents the "or" composite rule based on the specification pattern</summary>

        this.isSatisfied = ko.computed(function () {
            return leftHandSide.isSatisfied() ||
                rightHandSide.isSatisfied();
        });

        this.evaluate = function (instance) {
            leftHandSide.evaluate(instance);
            rightHandSide.evaluate(instance);
        };
    })
});

File: Source/Bifrost/Specifications/Or.cs
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) 2008-2017 Dolittle. All rights reserved.
 *  Licensed under the MIT License. See LICENSE in the project root for license information.
 *--------------------------------------------------------------------------------------------*/
using System.Linq.Expressions;

namespace Bifrost.Specifications
{
    /// <summary>
    /// Composes a rule that will be satisfied if either the first rule or second rule is satisfied
    /// </summary>
    /// <typeparam name="T">Type that the rule is to be evaluated for.</typeparam>
    /// <remarks>Based on http://bloggingabout.net/blogs/dries/archive/2011/09/29/specification-pattern-continued.aspx </remarks>
    internal class Or<T> : Specification<T>
	{
		internal Or(Specification<T> lhs, Specification<T> rhs)
		{
			Predicate = lhs.Compose(rhs, Expression.Or).Predicate;
		}
	}
}